examples/event/time/main.go

```diff
@@ -0,0 +1,199 @@
+}
+
+func (t *timer) getEndpoint() interface{} {
+	return "to-be-implemented:the url endpoint"
+}
+
+func (t *timer) init() *timer {
+	t.topics = map[string]interface{}{}
+
+	for _, topic := range types.PathFromStrings(
+		"msec/100",
+		"msec/500",
+		"sec/1",
+		"sec/5",
+		"sec/30",
+		"min/1",
+		"min/5",
+		"min/30",
+		"hr/1",
+	) {
+		types.Put(topic, t.getEndpoint, t.topics)
+	}
+
+	return t
+}
+
+// List returns the nodes under the given topic
+func (t *timer) List(topic types.Path) ([]string, error) {
+	return types.List(topic, t.topics), nil
+}
+
+// PublishOn sets the channel to publish on
+func (t *timer) PublishOn(c chan<- *event.Event) {
+	go func() {
+		defer close(c)
+
+		log.Infoln("Timer starting publish on channel:", c)
+
+		tick100msec := time.Tick(100 * time.Millisecond)
+		tick500msec := time.Tick(500 * time.Millisecond)
+		tick1sec := time.Tick(1 * time.Second)
+		tick5sec := time.Tick(5 * time.Second)
+		tick30sec := time.Tick(30 * time.Second)
+		tick1min := time.Tick(1 * time.Minute)
+		tick5min := time.Tick(5 * time.Minute)
+		tick30min := time.Tick(30 * time.Minute)
+		tick1hr := time.Tick(1 * time.Hour)
+
+		for {
+			select {
+			case <-t.stop:
+				return
+
+			case <-tick100msec:
+				c <- event.Event{
+					Type: timerType,
+					ID:   "100ms",
+				}.Init().Now().WithTopic("msec/100").WithDataMust(time.Duration(100 * time.Millisecond))
+
+			case <-tick500msec:
+				c <- event.Event{
+					Type: timerType,
+					ID:   "500ms",
+				}.Init().Now().WithTopic("msec/500").WithDataMust(time.Duration(500 * time.Millisecond))
+
+			case <-tick1sec:
+				c <- event.Event{
+					Type: timerType,
+					ID:   "1sec",
+				}.Init().Now().WithTopic("sec/1").WithDataMust(time.Duration(1 * time.Second))
+
+			case <-tick5sec:
+				c <- event.Event{
+					Type: timerType,
+					ID:   "5s",
+				}.Init().Now().WithTopic("sec/5").WithDataMust(time.Duration(5 * time.Second))
+
+			case <-tick30sec:
+				c <- event.Event{
+					Type: timerType,
+					ID:   "30s",
+				}.Init().Now().WithTopic("sec/30").WithDataMust(time.Duration(30 * time.Second))
+
+			case <-tick1min:
+				c <- event.Event{
+					Type: timerType,
+					ID:   "1m",
+				}.Init().Now().WithTopic("min/1").WithDataMust(time.Duration(1 * time.Minute))
+
+			case <-tick5min:
+				c <- event.Event{
+					Type: timerType,
+					ID:   "5m",
+				}.Init().Now().WithTopic("min/5").WithDataMust(time.Duration(5 * time.Minute))
+
+			case <-tick30min:
+				c <- event.Event{
+					Type: timerType,
+					ID:   "30m",
+				}.Init().Now().WithTopic("min/30").WithDataMust(time.Duration(30 * time.Minute))
+
+			case <-tick1hr:
+				c <- event.Event{
+					Type: timerType,
+					ID:   "1h",
+				}.Init().Now().WithTopic("hr/1").WithDataMust(time.Duration(1 * time.Hour))
+
+			}
+		}
+	}()
+}
+
+func main() {
+
+	cmd := &cobra.Command{
+		Use:   os.Args[0],
+		Short: "Timer event plugin",
+	}
+
+	name := cmd.Flags().String("name", "time", "Plugin name to advertise for discovery")
```

Comment:
As a trivial thing, I think that it is better to name it `event-time` in order to keep it in balance with other examples.